src/prefect/server/events/triggers.py

```diff
@@ -191,14 +238,33 @@ async def evaluate(
             return None
 
         start = pendulum.instance(max(bucket.end, now))
-        return await start_new_bucket(
-            session,
-            trigger,
-            bucketing_key=bucket.bucketing_key,
-            start=start,
-            end=start + trigger.within,
-            count=count,
-        )
+        end = start + trigger.within
+
+        # If we're processing a reactive trigger and leaving the function with a count
+        # that we've just spent in the bucket for the next time window, it means that we
+        # just processed an event that was in the future. It's possible that this event
+        # was sufficient enough to cause the trigger to fire, so we need to evaluate one
+        # more time to see if that's the case.  This is case #4 from the implementation
+        # notes above.
+        if trigger.posture == Posture.Reactive and count > 0:
+            bucket = await start_new_bucket(
+                session,
+                trigger,
+                bucketing_key=bucket.bucketing_key,
+                start=start,
+                end=end,
+                count=0,
+            )
+            return await evaluate(session, trigger, bucket, now, triggering_event)
```

Comment:
This is the heart of the change: if we're leaving this function with an "unspent" `count`, we'll make the new bucket and also re-`evaluate`.